### PR TITLE
[UIK-4274][feature-highlight] updated addon logic for Pills

### PR DESCRIPTION
**Pull request by slizhevskyv-semrush**
## Motivation and Context

Addon logic in `PillsFH` worked incorrectly. The fix ensures correct behavior in edge cases.

## How has this been tested?

I've added a browser test.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [x] I have added new tests on added of fixed functionality.

### CODE REVIEW

File: semcore/feature-highlight/__tests__/feature-highlight.browser-test.tsx [hunk 1]
```diff
@@ -58,44 +58,62 @@ test.describe('Feature highlight', () => {
...
-      { disabled: true, size: 'l', animatedSparkleCount: 0 },
-    ];
-
-    variables.forEach((item) => {
-      test(`Verify Pills ${item.disabled} ${item.size}`, async ({ page }) => {
-        const standPath = 'stories/patterns/ux-patterns/feature-highlight/advanced/examples/all-controls.tsx';
-        const htmlContent = await e2eStandToHtml(standPath, 'en', item);
-
-        await page.setContent(htmlContent);
+  test.describe('Pills', () => {
```
Comment (slizhevskyv-semrush) [inline semcore/feature-highlight/__tests__/feature-highlight.browser-test.tsx]: @Valeria-Zimnitskaya I've changed the structure slightly. Feel free to apply any changes to that :) Thanks!

File: semcore/feature-highlight/src/components/pills/Pills.tsx [hunk 2]
```diff
@@ -49,10 +50,10 @@ class HighlightedItemRoot extends Component {
 }
 
 function HighlightedItemAddon(props: HighlightedItemAddonProps & { clicked: boolean } & IRootComponentProps) {
-  const { clicked, animatedSparkleCount, Children, children: hasChildren } = props;
+  const { clicked, animatedSparkleCount, Children, children } = props;
   return (
     <Root render={Pills.Item.Addon}>
-      {hasChildren
+      {isNode(children)
```
Comment (ilyabrower) [inline semcore/feature-highlight/src/components/pills/Pills.tsx]: I think here will be incorrect check... with empty string for example here will be false
Let's check that Number(children) is non NaN and render sparkles, otherwise - <Children/>

Comment (slizhevskyv-semrush) [reply semcore/feature-highlight/src/components/pills/Pills.tsx]: good point, thanks!

**Review by Valeria-Zimnitskaya — APPROVED** [review]
